### ISSUE TITLE
Multiple aspects running py_binary based tools errors in Bazel 8

### ISSUE THREAD

**Issue by mark-thm** [bazelbuild/bazel#24619]
### Description of the bug:

Upon upgrading to Bazel 8.0.0, I started encountering an issue where two (or more) aspects that run a rules_python 1.0.0 py_binary as the executable are missing files. I see errors in the Python launcher script where the runfiles symlinks ultimately resolve to paths that do not, in fact, exist. The runfiles symlinks appear in the correct locations in the runfiles tree and are complete, but the locations the links point to are missing all but one of the aspect's underlying files.

<details>
<summary>More details/error messages:</summary>

When running a build I see errors like this:
```
ERROR: /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_pydeps++reqs+reqs/BUILD.bazel:7:15: Action external/rules_pydeps++reqs+reqs/pip_deps_index failed: (Exit 1): index failed: error executing Action command (from target @@rules_pydeps++reqs+reqs//:pip_deps_index) bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index ... (remaining 1 argument skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/2/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 580, in <module>
    Main()
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/2/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 504, in Main
    assert os.path.exists(main_filename), \
AssertionError: Cannot exec() '/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/2/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/private/index/index.py': file not found.
```

If I run with --sandbox_debug, and I inspect the sandbox, what I find is that the runfiles tree for index.runfiles looks like it should have the files:

```
❯ tree /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/6/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/
/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/6/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/
├── __init__.py
└── private
    ├── __init__.py
    ├── index
    │   ├── __init__.py -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/rules_pydeps+/pydeps/private/index/__init__.py
    │   ├── _index -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/_index
    │   ├── index -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index
    │   └── index.py -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/rules_pydeps+/pydeps/private/index/index.py
    └── py
        ├── __init__.py -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/rules_pydeps+/pydeps/private/py/__init__.py
        ├── python_module.py -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/rules_pydeps+/pydeps/private/py/python_module.py
        ├── python_source.py -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/rules_pydeps+/pydeps/private/py/python_source.py
        └── source_files.py -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/rules_pydeps+/pydeps/private/py/source_files.py

4 directories, 10 files
```

But when I inspect the execroot directories where the symlinks point, there is no `rules_pydeps+` directory:

```
❯ tree -L 1 /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/
/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/execroot/_main/external/
├── bazel_tools -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/bazel_tools
├── bazel_tools+xcode_configure_extension+local_config_xcode -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/bazel_tools+xcode_configure_extension+local_config_xcode
├── platforms -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/platforms
├── rules_cc+ -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_cc+
├── rules_cc++cc_configure_extension+local_config_cc -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_cc++cc_configure_extension+local_config_cc
├── rules_multitool++multitool+multitool -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_multitool++multitool+multitool
├── rules_multitool++multitool+multitool.macos_arm64 -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_multitool++multitool+multitool.macos_arm64
├── rules_mypy+ -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_mypy+
├── rules_python+ -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python+
├── rules_python++pip+pypi -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++pip+pypi
├── rules_python++pip+pypi_312_click_py3_none_any_ae74fb96 -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++pip+pypi_312_click_py3_none_any_ae74fb96
├── rules_python++pip+rules_mypy_pip -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++pip+rules_mypy_pip
├── rules_python++pip+rules_mypy_pip_312_mypy_cp312_cp312_macosx_11_0_arm64_39bb21c6 -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++pip+rules_mypy_pip_312_mypy_cp312_cp312_macosx_11_0_arm64_39bb21c6
├── rules_python++pip+rules_mypy_pip_312_mypy_extensions_py3_none_any_4392f6c0 -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++pip+rules_mypy_pip_312_mypy_extensions_py3_none_any_4392f6c0
├── rules_python++pip+rules_mypy_pip_312_typing_extensions_py3_none_any_04e5ca03 -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++pip+rules_mypy_pip_312_typing_extensions_py3_none_any_04e5ca03
├── rules_python++python+python_3_12_aarch64-apple-darwin -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_python++python+python_3_12_aarch64-apple-darwin
└── rules_uv+ -> /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_uv+
```

It does looks like `/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_pydeps+`, so the process that constructs execroot seems to be missing a relevant link for some reason.
</details>

### Which category does this issue belong to?

Local Execution

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

A small but perhaps not minimal repro here: https://github.com/theoremlp/bazel8_problem_repro

Changing .bazelversion to 7.4.1 and running `bazel build ...` succeeds. Running `bazel build ...` at version 8.0.0 fails with symlink problems as described above.

### Which operating system are you running Bazel on?

MacOS, Ubuntu

### What is the output of `bazel info release`?

release 8.0.0

### If this is a regression, please try to identify the Bazel commit where the bug was introduced with bazelisk --bisect.

(see comment below, the initial bisect did not correctly identify the first bad commit)

### Have you found anything relevant by searching the web?

Unfortunately no.

**Comment by fmeum** [comment]
The bisect to https://github.com/bazelbuild/bazel/commit/1744d45c0c57af23f631c2145e8af4b185dca77f is extremely surprising. Could you verify manually that this passes on the patent commit and fails on the commit itself after a `bazel clean --expunge` each time?

**Comment by fmeum** [comment]
@bazel-io flag

**Comment by mark-thm** [comment]
Yeah I’m finding strange bisect results, so going to redo the bisect
manually.

***@***.*** seems to pass though. So it’s something specific to the 8.0.0
release branch, I think.

-- 










This communication, together with any attachments, is intended 
only for the addressee(s) and may contain confidential, privileged or 
proprietary information of Theorem Partners LLC ("Theorem"). By accepting 
this communication you agree to keep confidential all information contained 
in this communication, as well as any information derived by you from the 
confidential information contained in this communication. Theorem does not 
waive any confidentiality by misdelivery.

If you receive this 
communication in error, any use, dissemination, printing or copying of all 
or any part of it is strictly prohibited; please destroy all electronic and 
paper copies and notify the sender immediately. Nothing in this email is 
intended to constitute (1) investment, legal or tax advice, (2) any 
recommendation to purchase or sell any security, (3) any advertisement or 
offer of advisory services or (4) any offer to sell or solicitation of an 
offer to buy any securities or other financial instrument in any 
jurisdiction.

Theorem, including its agents or affiliates, reserves the 
right to intercept, archive, monitor and review all communications to and 
from its network, including this email and any email response to it.

Theorem makes no representation as to the accuracy or completeness of the 
information in this communication and does not accept liability for any 
errors or omissions in this communication, including any liability 
resulting from its transmission by email, and undertakes no obligation to 
update any information in this email or its attachments.

**Comment by mark-thm** [comment]
Ok, I've re-run backwards from 8.0.0 and the first commit where the build fails is 27487cf11f61eb23108f8df129dc1a00ee1a4370.

Verifying like this:
```
❯ USE_BAZEL_VERSION=e749da2f60025e2462f7cc7eea0457eb5734a51a bazelisk clean --expunge
INFO: Starting clean (this may take a while). Use --async if the clean takes more than several minutes.
❯ USE_BAZEL_VERSION=e749da2f60025e2462f7cc7eea0457eb5734a51a bazelisk build ...
Starting local Bazel server and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
INFO: Analyzed 5 targets (107 packages loaded, 6240 targets configured).
INFO: Found 5 targets...
INFO: Elapsed time: 26.112s, Critical Path: 17.74s
INFO: 40 processes: 37 internal, 3 darwin-sandbox.
INFO: Build completed successfully, 40 total actions

❯ USE_BAZEL_VERSION=27487cf11f61eb23108f8df129dc1a00ee1a4370 bazelisk clean --expunge
Starting local Bazel server and connecting to it...
INFO: Starting clean (this may take a while). Use --async if the clean takes more than several minutes.
❯ USE_BAZEL_VERSION=27487cf11f61eb23108f8df129dc1a00ee1a4370 bazelisk build ...
Starting local Bazel server and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
INFO: Analyzed 5 targets (107 packages loaded, 6240 targets configured).
ERROR: /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_pydeps++reqs+reqs/BUILD.bazel:7:15: Action external/rules_pydeps++reqs+reqs/pip_deps_index failed: (Exit 1): index failed: error executing Action command (from target @@rules_pydeps++reqs+reqs//:pip_deps_index) bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index ... (remaining 1 argument skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 580, in <module>
    Main()
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 504, in Main
    assert os.path.exists(main_filename), \
AssertionError: Cannot exec() '/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/private/index/index.py': file not found.
INFO: Elapsed time: 14.971s, Critical Path: 2.08s
INFO: 39 processes: 39 internal.
ERROR: Build did NOT complete successfully
```

**Comment by mark-thm** [comment]
I think when I tested this on master originally it was succeeding because of a cache. It still appears to be failing on master, though:
```
❯ USE_BAZEL_VERSION=b09335b86e03732d6da2fe40a7ac07af48480771 bazelisk clean --expunge
Starting local Bazel server (no_version) and connecting to it...
INFO: Starting clean (this may take a while). Use --async if the clean takes more than several minutes.
❯ USE_BAZEL_VERSION=b09335b86e03732d6da2fe40a7ac07af48480771 bazelisk build ...
Starting local Bazel server (no_version) and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
INFO: Analyzed 5 targets (107 packages loaded, 6240 targets and 26 aspects configured).
ERROR: /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_pydeps++reqs+reqs/BUILD.bazel:7:15: Action external/rules_pydeps++reqs+reqs/pip_deps_index failed: (Exit 1): index failed: error executing Action command (from target @@rules_pydeps++reqs+reqs//:pip_deps_index) bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index ... (remaining 1 argument skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 580, in <module>
    Main()
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 504, in Main
    assert os.path.exists(main_filename), \
AssertionError: Cannot exec() '/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/private/index/index.py': file not found.
INFO: Elapsed time: 15.820s, Critical Path: 2.02s
INFO: 47 processes: 47 internal.
ERROR: Build did NOT complete successfully
```

**Comment by iancha1992** [comment]
@bazel-io fork 8.1.0

**Comment by mark-thm** [comment]
On master, the most recent 'good' commit was eae9d4d019ca8d01bd3db5a8549b7f0e3505db61, and the subsequent 'bad' commit was 54237be06080d16a50f9c9ab0ad3a8f2651ca4d2.

```
❯ USE_BAZEL_VERSION=eae9d4d019ca8d01bd3db5a8549b7f0e3505db61 bazelisk clean --expunge
Starting local Bazel server (no_version) and connecting to it...
INFO: Starting clean (this may take a while). Use --async if the clean takes more than several minutes.
❯ USE_BAZEL_VERSION=eae9d4d019ca8d01bd3db5a8549b7f0e3505db61 bazelisk build ...
Starting local Bazel server (no_version) and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
INFO: Analyzed 5 targets (107 packages loaded, 6240 targets and 26 aspects configured).
INFO: Found 5 targets...
INFO: Elapsed time: 25.012s, Critical Path: 14.97s
INFO: 48 processes: 45 internal, 3 darwin-sandbox.
INFO: Build completed successfully, 48 total actions
```

```
❯ USE_BAZEL_VERSION=54237be06080d16a50f9c9ab0ad3a8f2651ca4d2 bazelisk clean --expunge
Starting local Bazel server (no_version) and connecting to it...
INFO: Starting clean (this may take a while). Use --async if the clean takes more than several minutes.
❯ USE_BAZEL_VERSION=54237be06080d16a50f9c9ab0ad3a8f2651ca4d2 bazelisk build ...
Starting local Bazel server (no_version) and connecting to it...
WARNING: Couldn't auto load rules or symbols, because no dependency on module/repository 'rules_android' found. This will result in a failure if there's a reference to those rules or symbols.
INFO: Analyzed 5 targets (107 packages loaded, 6240 targets and 26 aspects configured).
ERROR: /private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/external/rules_pydeps++reqs+reqs/BUILD.bazel:7:15: Action external/rules_pydeps++reqs+reqs/pip_deps_index failed: (Exit 1): index failed: error executing Action command (from target @@rules_pydeps++reqs+reqs//:pip_deps_index) bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index ... (remaining 1 argument skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
Traceback (most recent call last):
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 580, in <module>
    Main()
  File "/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index", line 504, in Main
    assert os.path.exists(main_filename), \
AssertionError: Cannot exec() '/private/var/tmp/_bazel_melliot/ec7f0e67bfc3fe3a9bd546beff97b5c5/sandbox/darwin-sandbox/1/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/private/index/index.py': file not found.
INFO: Elapsed time: 12.961s, Critical Path: 2.29s
INFO: 47 processes: 47 internal.
ERROR: Build did NOT complete successfully
```

**Comment by fmeum** [comment]
@bazel-io fork 8.0.1

**Comment by fmeum** [comment]
This is extremely confusing: On the Bazel 8 release branch, the reproducer fails with https://github.com/bazelbuild/bazel/commit/27487cf11f61eb23108f8df129dc1a00ee1a4370 and passes with its parent. On the master branch, it does fail with https://github.com/bazelbuild/bazel/commit/1744d45c0c57af23f631c2145e8af4b185dca77f and pass with its parent for me after a clean.

**Comment by fmeum** [comment]
When I run this with `--sandbox_debug`, I see:

```
ls -lah /private/var/tmp/_bazel_fmeum/e78feca5b66fe0e6f6d23e5ae0702f4d/sandbox/darwin-sandbox/10/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/private/index/index.py
lrwxr-xr-x@ 1 fmeum  wheel   130B Dec 12 21:03 /private/var/tmp/_bazel_fmeum/e78feca5b66fe0e6f6d23e5ae0702f4d/sandbox/darwin-sandbox/10/execroot/_main/bazel-out/darwin_arm64-opt-exec-ST-c6280171b4e4/bin/external/rules_pydeps+/pydeps/private/index/index.runfiles/_main/../rules_pydeps+/pydeps/private/index/index.py -> /private/var/tmp/_bazel_fmeum/e78feca5b66fe0e6f6d23e5ae0702f4d/execroot/_main/external/rules_pydeps+/pydeps/private/index/index.py
```

The target of the symlink (in fact, the entire `external/rules_pydeps+` directory) is missing when the build fails. It's possible that this is happening non-deterministically, which throws off our bisect. This looks similar to past issues with Skymeld.

**Comment by fmeum** [comment]
Turns out that the event deduplication in `BuildDriverFunction` is at fault here since it deduplicates per target, not per aspect applied to each target. I sent https://github.com/bazelbuild/bazel/pull/24676 to fix this. 

The reproducer doesn't apply at HEAD because other commits changed the order in which the two aspects are evaluated, making it so that the aspect with the failing action would now come first and thus be the one to have its symlinks planted.

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.1 RC1](https://github.com/bazelbuild/bazel/releases/tag/8.0.1rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.1rc1. Thanks!
